aireview/src/aireview/prefect/deploy/manage_flow.py

```diff
@@ -0,0 +1,176 @@
+from prefect.client.orchestration import PrefectClient  # noqa
+from prefect.client.schemas.filters import (
+    DeploymentFilter,
+    DeploymentFilterTags,
+    FlowRunFilter,
+)  # noqa
+
+
+def env_to_workpool(env: str):
+    """
+    Return the workpool name based on the environment
+    """
+    low_cased_env = env.lower()
+    if low_cased_env == "prd":
+        return "workpool-k8s-prd"
+    elif low_cased_env == "dev":
+        return "workpool-k8s-dev"
+    else:
+        return "workpool-k8s-rap"
+
+
+async def deploy(
+    flow: Flow,
+    product: str,
+    env: str,
+    parameters: dict = {},
+    additional_job_variables: dict = {},
+    concurrency_limit: int = 3,
+    cron: str | None = None,
+    work_queue_name: str = "default",
+    image_tag: str = "test-local",
+    name_override: str | None = None,
+):
+    """
+    This is the function that will deploy the flows to the Prefect server
+    Add your flow deployment here.
+
+    flow: The flow function
+    product: The product name (pdx, dpf, kamai)
+    env: The environment (prd, dev, prxxxxx)
+    parameters: The parameters for the flow
+    additional_job_variables: Additional job variables to be passed to the flow
+    concurrency_limit: The concurrency limit for the flow
+    cron: The cron schedule for the flow
+    work_queue_name: The work queue name for the flow
+    image_tag: The image tag for the flow
+    name_override: The name override for the flow
+    """
+    result = await flow.with_options(name=f"{flow.name}-{env}").deploy(
+        name=f"{product}-{env}" if name_override is None else name_override,
+        image=f"crdataprd.azurecr.io/prefect-flow-{product}:{image_tag}".lower(),
+        work_pool_name=env_to_workpool(env),
+        work_queue_name=work_queue_name,
+        cron=cron,
+        tags=[env.lower(), product.lower()],
+        job_variables={
+            "env": {
+                f"{product.upper()}_ENV": env,
+            },
+            **additional_job_variables,
+        },
+        parameters=parameters,
+        concurrency_limit=concurrency_limit,
+        entrypoint_type=EntrypointType.MODULE_PATH,
+        build=False,
+    )
+    print(f"Deployment ID: {result}")
+    return result
+
+
+async def remove_all_resource_for_pr(env: str):
+    """
+    Remove all resources for a PR:
+    1. snowflake credentials block
+    2. deployments and its runs
+    """
+    client = get_client()
+    # remove the snowflake credentials block
+
+    print("💥 Deleting the deployments and its runs")
+    deployment_filter = DeploymentFilter(tags=DeploymentFilterTags(all_=[env.lower()]))
+
+    print(
+        "Fetching deployments with filter: {}".format(
+            deployment_filter.__dict__["tags"].__dict__["all_"]
+        )
+    )
+    deployments = await client.read_deployments(deployment_filter=deployment_filter)
+
+    for d in deployments:
+        flow = await client.read_flow(d.flow_id)
```

Comment:
**documentation**: Comment '# remove the snowflake credentials block' suggests incomplete implementation. Either implement the functionality or remove the comment.